### PR TITLE
Raise the correct error when quay returns 403 in get_releases_raw()

### PR DESCRIPTION
**Pull request by chmeliik**
Signed-off-by: Adam Cmiel <acmiel@redhat.com>

### CODE REVIEW

**Comment by csomh** [comment]
Is this possible to be tested on integration level? Do the integration tests need to be updated?

If no, please make a note in the commit message on why this doesn't require such an update, so that future readers don't have to dig the code to find it out :)

File: omps/quay.py [hunk 1]
```diff
@@ -269,22 +270,27 @@ def get_releases_raw(self, repo):
...
-        except requests.exceptions.HTTPError as http_e:
-            if http_e.response.status_code == 404:
-                raise QuayPackageNotFound(
-                    "Package {}/{} not found".format(
-                        self._organization, repo
-                    )
-                )
-            _raise(http_e)
-        except requests.exceptions.RequestException as e:
+        except (requests.exceptions.HTTPError,
```
Comment (MartinBasti) [inline omps/quay.py]: Isn't `HTTPError` subclass of `RequestExceptions`?

http://docs.python-requests.org/en/master/_modules/requests/exceptions/

**Comment by chmeliik** [comment]
@csomh It definitely should be possible, after taking a quick look at the current tests, these seem to be the ones that will need to be updated:
```python
delete_version.test_organization_unaccessible_in_quay()
push_archive_test.test_organization_unaccessible_in_quay()
# not sure about the second one as the expected message doesn't say
# anything about authorization, but it looks a lot like the first one otherwise
```
Generally, it would be any tests where the the expected error is a `QuayPackageError` caused by authorization issues (`QuayPackageError` is being replaced by `QuayAuthorizationError`, and the status code is changing from `500` to `403`)

File: omps/quay.py (hunk 1 above)
Comment (chmeliik) [reply omps/quay.py]: You're right, it used to have separate error handling logic but that is now in `_raise()`, will fix

**Comment by chmeliik** [comment]
@csomh On second thought, the errors in those two methods are likely unrelated to the `get_releases_raw()` method, in which case they would not need changing at all.

**Comment by csomh** [comment]
@chmeliik, yeah, that was my first impression also... but after this discussion probably we are missing a test where the token provided to OMPS is invalid. I'll double-check this and add one if needed.

**Comment by chmeliik** [comment]
@csomh Should I add that comment about why integration tests don't need to be changed then?

**Comment by chmeliik** [comment]
updated PR, added note about integration tests to commit message and removed that extra exception in `except ...`

**Pull request merged**